### ISSUE TITLE
xenops-image-handle-paste not working in linux! 

### ISSUE THREAD

**Issue by OmarAshkar** [dandavison/xenops#24]
Hi,
I am trying to the function xenops-image-hande-paste in OS. I am using xfce, doom emacs, xclip 0.13 and xfce4-screenshot. 

I got no messages or errors, just nothing!

Any help is highly appreciated.

Thanks!

**Comment by dandavison** [comment]
Hi @OmarAshkar. Could you tell me the version of magit-delta that you have installed? I don't have a linux machine to hand to test this with your setup, but this is _supposed_ to be working; see my comment here: https://github.com/dandavison/xenops/issues/19#issuecomment-753614843

**Comment by OmarAshkar** [comment]
I have reviewed your code quickly and I think it should work. I will try to debug when I have time. Thank you!

**Comment by OmarAshkar** [comment]
Ok, on my debugging I found that this let chuck does not get executed at all. 

https://github.com/dandavison/xenops/blob/4994ae4a660ee94d341ce1905c12b4cf39fee574/lisp/xenops-image.el#L104-L112

I got the image written to me tmp directory, but no prompt for a name or anything!

**Comment by dandavison** [comment]
Is it a problem with emacs not finding your xclip executable? What do you get if you evaluate `(executable-find "xclip")` in emacs? If `nil` then perhaps your `exec-path` needs to be adjusted so that it picks up the directory containing your xclip executable.

**Comment by OmarAshkar** [comment]
No, xclip function is evaluated and temporary image is written in /tmp/ directory. I debugged the code line by line, the function `xenops-image-suggest-file-name` was never invoked!

**Comment by dandavison** [comment]
> Ok, on my debugging I found that this let chuck does not get executed at all.

OK, so it must be the case that both these return `nil`, correct? https://github.com/dandavison/xenops/blob/4994ae4a660ee94d341ce1905c12b4cf39fee574/lisp/xenops-image.el#L102-L103

If so, that implies that either `(executable-find "xclip")` is nil, or the exit status of the `xclip` subprocess is non-zero. Is thatg correct so far?

**Comment by OmarAshkar** [comment]
You are right on that. I found that the image is empty! (executable-find "xclip") is NOT nil.  
I think it's the exit-status! 

And no I don't have pngpaste.  I don't know if you have any ideas!

**Comment by dandavison** [comment]
OK cool, I think we should be able to get to the bottom of this then. What I would do is run the xclip command from the command line, outside emacs, and see what's going wrong. (I'm sure you know, in bash you can check the exit status with `echo $?`). So you want to run the command that this lisp code corresponds to:

```emacs-lisp
(call-process "xclip" nil `(:file ,temp-file) nil "-t" "image/png" "-o")
```

It sounds like you're using `edebug`, so perhaps you could edebug that function, and just before emacs evaluates that line, find the value of the variable `temp-file`, and then switch to a terminal and run the xclip command on whatever `temp-file` is in the shell.

**Comment by dandavison** [comment]
More generally, outside emacs, can you copy image data to the clipboard and confirm that you can use `xclip -t image/png -o > img.png` to write the image to disk?

(That works for me on MacOS, after I start a terminal using the program XQuartz.)

**Comment by OmarAshkar** [comment]
Right, Thanks! I have fixed the xclip flags and corrected in this PR #25. 

Check it out! Thanks!

**Comment by dandavison** [comment]
Thanks a lot. I've merged that and it should be automatically picked up by MELPA in the next few hours.
